source/slang/slang-capability.h

```diff
@@ -46,108 +47,34 @@ namespace Slang
-
-        /// Is this capability set empty (such that any target supports it)?
-    bool isEmpty() const;
-
-        /// Is this capability set invalid (such that no target could support it)?
-    bool isInvalid() const;
-
-    // Capabilities are "incompatible" if no target platform can ever support both
-    // at the same time. For example, the `HLSL` and `GLSL` capabilities are
-    // incompatible, because a single target cannot be both an HLSL target and
-    // a GLSL target (at least for now).
-    //
-    // Note that we are using the term "incompatible" here even though it
-    // seems like "disjoint" would be intuitively correct (HLSL and GLSL
-    // targets sure do seem to be disjoint). The problem is that in our
-    // set-theoretic representation of capabilities, incompatible capability
-    // sets are *never* disjoint sets of atoms, and (valid) disjoint sets of atoms
-    // *never* represent incompatible capability sets.
-
-        /// Is this capability set incompatible with the given `other` set.
-    bool isIncompatibleWith(CapabilityAtom other) const;
-
-        /// Is this capability set incompatible with the given `other` atomic capability.
-    bool isIncompatibleWith(CapabilityConjunctionSet const& other) const;
-
-    // One capability set A "implies" another set B if a target that
-    // supports A must also support all of B.
-    //
-    // In practice, this means that "A implies B" is the same as
-    // "A is a subset of B" in the set-theoretic model, but
-    // we ant to think of this primarily as supported/required features,
-    // and not get hung up on the set theory.
-
-        /// Does this capability set imply all the capabilities in `other`?
-    bool implies(CapabilityConjunctionSet const& other) const;
-
-
-        /// Does this capability set imply the atomic capability `other`?
-    bool implies(CapabilityAtom other) const;
-
-    // A capability set is equal to another if each implies the other.
-
-        /// Are these two capability sets equal?
-    bool operator==(CapabilityConjunctionSet const& that) const;
-    bool operator<(CapabilityConjunctionSet const& that) const;
+struct CapabilityTargetSet;
+typedef Dictionary<CapabilityAtom, CapabilityTargetSet> CapabilityTargetSets;
 
-        /// Get access to the raw atomic capabilities that define this set.
-    List<CapabilityAtom> const& getExpandedAtoms() const { return m_expandedAtoms; }
-    List<CapabilityAtom>& getExpandedAtoms() { return m_expandedAtoms; }
-
-        /// Calculate a list of "compacted" atoms, which excludes any atoms from the expanded list that are implies by another item in the list.
-    void calcCompactedAtoms(List<CapabilityAtom>& outAtoms) const;
+/// CapabilityStageSet encapsulates all capabilities of a specific shader stage for a specific target.
+/// Capabilities may be disjoint, but only in rare cases: 
+/// {{glsl, _GLSL_130, GL_EXT_FOO1}, {glsl, _GLSL_130, _GLSL_140, _GLSL_150}}
```

Comment:
> Can we avoid having such kind of disjoint cases?
The point of disjoint sets is that they are different ways to use the same hardware features. I suppose we don't need support for this (just use one or the other).

Reply:
My thinking is that we don't want to allow this. Because this can lead to an explosion of disjoint cases if a function uses N features that may be supported differently, you will get 2^N disjoint sets.

Using inheritance/implication relationship I described will keep things simple and can still work for our case.

Reply:
disjointSets are now disallowed 👍

`addNewSet` controls this behavior.